oak_functions/loader/src/attestation.rs

```diff
@@ -186,64 +181,8 @@ where
     }
 }
 
-/// Attest a single gRPC streaming request. Client messages are provided via `receiver`.
-async fn process_client_hello(
-    receiver: &mut Receiver,
-    tee_certificate: &[u8],
-) -> anyhow::Result<(AttestedInvokeResponse, ServerAttestationEngine<Attesting>)> {
-    let request = receiver
-        .receive()
-        .await
-        .context("Couldn't receive attestation request")?
-        .context("Stream stopped preemptively")?;
-
-    // Receive client hello message.
-    let request_type = request.request_type.context("Couldn't read request type")?;
-    let client_hello = if let RequestType::AttestationMessage(request) = request_type {
-        request.body
-    } else {
-        anyhow::bail!("Received incorrect message type");
-    };
-
-    let attestation_engine = ServerAttestationEngine::<Initializing>::new(
-        AttestationBehavior::create_self_attestation(tee_certificate)
-            .context("Couldn't create self attestation behavior")?,
-    );
-    let (server_identity, attestation_engine) = attestation_engine
-        .process_client_hello(&client_hello)
-        .context("Couldn't accept client hello")?;
-
-    // Create server attestation identity.
-    let attestation_response = AttestedInvokeResponse {
-        response_type: Some(ResponseType::AttestationMessage(AttestationMessage {
-            body: server_identity,
-        })),
-    };
-    Ok((attestation_response, attestation_engine))
-}
-
-/// Attest a single gRPC streaming request. Client messages are provided via `receiver`.
-async fn process_client_identity(
-    receiver: &mut Receiver,
-    attestation_engine: ServerAttestationEngine<Attesting>,
-) -> anyhow::Result<AeadEncryptor> {
-    let request = receiver
-        .receive()
-        .await
-        .context("Couldn't receive attestation request")?
-        .context("Stream stopped preemptively")?;
-
-    // Receive client attestation identity.
-    let request_type = request.request_type.context("Couldn't read request type")?;
-    let client_identity = if let RequestType::AttestationMessage(request) = request_type {
-        request.body
-    } else {
-        anyhow::bail!("Received incorrect message type");
-    };
-
-    let attestation_engine = attestation_engine
-        .process_client_identity(&client_identity)
-        .context("Couldn't attest the client")?;
-    let encryptor = attestation_engine.get_encryptor();
-    Ok(encryptor)
+fn error_to_status<T: std::fmt::Debug>(comment: &str, error: T) -> Status {
+    let message = format!("{}: {:?}", comment, error);
+    warn!("{}", message);
+    Status::internal(message)
```

Comment:
The higher level function is a `async_stream::try_stream!`, and it's not possible to use `?` there.
So this function is just a code reduction: I use it in `map_err` instead of copy-pasting this code in every `map_err` statement.

Reply:
Can you fix the logging though?

Reply:
Fixed logging